.github/actions/setup-caddy/action.yml

```diff
@@ -0,0 +1,13 @@
+name: Setup Caddy server
+runs:
+  using: composite
+  steps:
+    - shell: bash
+      run: |
+        sudo apt-get install -y debian-keyring debian-archive-keyring apt-transport-https
+        curl -1sLf 'https://dl.cloudsmith.io/public/caddy/stable/gpg.key' | sudo gpg --dearmor -o /usr/share/keyrings/caddy-stable-archive-keyring.gpg
+        curl -1sLf 'https://dl.cloudsmith.io/public/caddy/stable/debian.deb.txt' | sudo tee /etc/apt/sources.list.d/caddy-stable.list
+        sudo apt-get update
+        sudo apt-get install -y caddy
+
+        sudo caddy start --config ext/curl/tests/Caddyfile
```

Comment:
This does add 1m30 to the build time for 2 tests...

Reply:
The slowest part is installing the package. Do you want me to try downloading the binary instead?

On the other hand, with a local webserver, we could probably unskip a lot of tests that are currently slow because they hit a remote server and then increase the overall reliability.

Reply:
> Do you want me to try downloading the binary instead?

Sure. Let's see how that compares.

> On the other hand, with a local webserver, we could probably unskip a lot of tests that are currently slow because they hit a remote server and then increase the overall reliability.

I'm not sure if these are skipped for speed or reliability. But Caddy would improve both of those though. But that shouldn't depend on how we install Caddy.

Reply:
Just 13s with the new method 🎉